src/components/input/input.scss

```diff
@@ -0,0 +1,182 @@
+@import 'default-theme';
+@import 'mixins';
+@import 'variables';
+
+
+// Placeholder colors. Required is used for the `*` star shown in the placeholder.
+$md-input-placeholder-color: md-color($md-foreground, hint-text);
+$md-input-floating-placeholder-color: md-color($md-primary);
+$md-input-required-placeholder-color: md-color($md-accent);
+
+// Underline colors.
+$md-input-underline-color: md-color($md-foreground, hint-text);
+$md-input-underline-color-accent: md-color($md-accent);
+$md-input-underline-disabled-color: md-color($md-foreground, hint-text);
+$md-input-underline-focused-color: md-color($md-primary);
+
+// Gradient for showing the dashed line when the input is disabled.
+$md-input-underline-disabled-background-image: linear-gradient(to right,
+        rgba(0,0,0,0.26) 0%, rgba(0,0,0,0.26) 33%, transparent 0%);
+
+:host {
+  display: inline-block;
+  position: relative;
+  font-family: $md-font-family;
+
+  // Global wrapper. We need to apply margin to the element for spacing, but
+  // cannot apply it to the host element directly.
+  .md-input-wrapper {
+    margin: 16px 0;
+  }
+
+  // We use a table layout to baseline align the prefix and suffix classes.
+  // The underline is outside of it so it can cover all of the elements under
+  // this table.
```

Comment:
Can you talk about why `display: inline-table` is more suitable for this than other approaches (especially flex)?